M/MLX_C/build_tarballs.jl

```diff
@@ -0,0 +1,78 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MLX_C"
+version = v"0.0.11" # This is an unofficial version number - used for git commit sha (126253...) matching MLX v0.20.0
+
+sources = [
+    GitSource("https://github.com/ml-explore/mlx-c.git", "126253627b95a0803273f44655fdaf5e7be4fbe1"),
+    ArchiveSource("https://github.com/roblabla/MacOSX-SDKs/releases/download/macosx14.0/MacOSX14.0.sdk.tar.xz",
+                  "4a31565fd2644d1aec23da3829977f83632a20985561a2038e198681e7e7bf49"),
+    DirectorySource("./bundled"),
+]
+
+script = raw"""
+# MLX hack
+cd $includedir
+mv mlx/io.h mlx/io.h.bak
+cp mlx/io.h.bak mlx/io.h
+mv mlx/backend/metal/metal.h mlx/backend/metal/metal.h.bak
+cp mlx/backend/metal/metal.h.bak mlx/backend/metal/metal.h
+atomic_patch -p1 $WORKSPACE/srcdir/patches/mlx-std_unordered_map.patch
+
+if [[ "$target" == *-apple-darwin* ]]; then
+    apple_sdk_root=$WORKSPACE/srcdir/MacOSX14.0.sdk
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" $CMAKE_TARGET_TOOLCHAIN
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" /opt/bin/$bb_full_target/$target-clang++
+fi
+
+cd $WORKSPACE/srcdir/mlx-c
+
+atomic_patch -p1 ../patches/cmake-win32.patch
+
+if [[ "$target" == *-w64-mingw32* ]]; then
+    atomic_patch -p1 ../patches/cmake-win32-io.patch
+fi
+
+export MACOSX_DEPLOYMENT_TARGET=13.3
```

Comment:
Shouldn't this go in the apple-darwin branch above?

Reply:
IIRC, the rationale is/was that it might be better, in general, to avoid branches, e.g. for patches - better to patch uniformly across all platforms - if possible.

… and in this case I guess this env. var. doesn’t matter for non-macOS platforms?

But happy to adjust.

Reply:
> IIRC, the rationale is/was that it might be better, in general, to avoid branches,

A branch is already there.